packages/core-api/src/app/configLoaders/composeConfigLoaders.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { AppConfig } from '@backstage/config';
+import { AppConfigLoader } from '../types';
+
+/**
+ * Executes config loaders and merges the results. The precedence is determined by the order they are supplied to the
+ * function (first parameter has highest priority).
+ *
+ * Where loaderA returns [appConfigA1, appConfigA2] and loaderB returns [appConfigB]
+ * await composeConfigLoaders(loaderA, loaderB) returns [appConfigA1,appConfigA2, appConfigB]
+ */
+export const composeConfigLoaders = (
```

Comment:
Tbh I think having this is overkill. Waiting for multiple loaders is pretty simple anyway and doesn't really require this abstraction. It's nice to have in the app, but the default one should at this point really cover 99%. If we do end up with a bunch of different config loaders that we want to re-use independently it might be worthwhile, but would rather keep this all private API for now.

Reply:
Fair enough 👍